includes/functions.php

```diff
@@ -262,11 +263,20 @@ function getImageName($device, $use_database = true, $dir = 'images/os/')
 function renamehost($id, $new, $source = 'console')
 {
     $host = gethostbyid($id);
+    $host_rrd_dir = Rrd::dirFromHost($host);
+    $is_remote_rrd = false;
 
-    if (! is_dir(Rrd::dirFromHost($new)) && rename(Rrd::dirFromHost($host), Rrd::dirFromHost($new)) === true) {
+    if (! is_dir($host_rrd_dir) && Rrd::checkRrdExists($host_rrd_dir)) { // We're most likely using a remote rrdcached
+        $is_remote_rrd = true;
+    }
+    if ($is_remote_rrd || (! is_dir(Rrd::dirFromHost($new)) && rename(Rrd::dirFromHost($host), Rrd::dirFromHost($new)) === true)) {
         dbUpdate(['hostname' => $new, 'ip' => null], 'devices', 'device_id=?', [$id]);
         log_event("Hostname changed -> $new ($source)", $id, 'system', 3);
 
+        if ($is_remote_rrd) {
+            throw new RemoteRrdRenameException('Host renamed but the associated RRD folder must be renamed manually');
```

Comment:
Wouldn't changing the hostname without renaming the files cause data loss? (and new files to be created next poll)

Reply:
Yes indeed, however currently the only way to rename a device when running with a remote rrdcached is to perform the SQL query manually **and** rename the folder on the rrdcached server.

The idea here is to remove the overly complicated step of manual SQL action, the folder renaming cannot be avoided but that's expected.